### PR TITLE
max_rules parameter now enforced in RuleFit 

### PR DESCRIPTION
**Pull request by keyan3**
- Enforce max_rules parameter via Lasso regularization param
- Remove ' ', '/', '<' from feature names since they create problems when querying DF

### CODE REVIEW

File: imodels/util/score.py [hunk 1]
```diff
@@ -63,19 +64,32 @@ def _eval_rule_perf(rule, X, y) -> Tuple[float, float]:
...
+    for alpha in alphas:
+        m = Lasso(alpha=alpha, random_state=random_state)
+        mse = 0
+        for train_index, test_index in kf.split(X):
+            X_train, X_test = X[train_index], X[test_index]
+            y_train, y_test = y[train_index], y[test_index]
+            m.fit(X_train, y_train)
+            mse += np.mean((m.predict(X_test) - y_test) ** 2)
+        
+        rule_count = sum(np.abs(m.coef_[-len(rules):]) > 0.001)
```
Comment (keyan3) [inline imodels/util/score.py:83]: This uses 0.001 as the threshold for when we consider a coef to be basically zero — lmk if it should just be 0

Comment (csinva) [reply imodels/util/score.py:83]: Good call, this shouldn't be exactly zero but we may want to tolerate small coefs when the values in y are themselves small or there is a big imbalance. Could you make the threshold `1e-6 / np.mean(np.abs(y_train)` ?

Comment (keyan3) [reply imodels/util/score.py]: Done

**Comment by csinva** [comment]
Looks great!

**Pull request merged**